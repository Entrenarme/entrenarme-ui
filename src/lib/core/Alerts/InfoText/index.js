// @flow
import * as React from 'react';
import styled from 'styled-components';

import colors from '../../../helpers/colors';
import { regularFont } from '../../../helpers/fonts';

const setColor = status => {
  switch (status) {
    case 'success':
      return colors.success.default;
    case 'warning':
      return colors.warning.default;
    default:
      return colors.error.default;
  }
};

const MainContainer = styled.div`
  display: flex;
  align-items: center;

  color: ${props => setColor(props.options.type)};
`;

const IconContainer = styled.div`
  display: inherit;
  margin-right: 10px;

  /* Smartphones (portrait) ----------- */
  @media only screen and (max-width: 320px) {
    margin-right: 8px;
  }

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (min-width: 320px) and (max-width: 767px) {
    margin-right: 8px;
  }
`;

const BodyContainer = styled.div`
  font-family: ${regularFont};
  font-size: 0.875rem;

  /* Smartphones (portrait) ----------- */
  @media only screen and (max-width: 320px) {
    font-size: 0.75rem;
  }

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (min-width: 320px) and (max-width: 767px) {
    font-size: 0.75rem;
  }
`;

type Props = {
  text: React.Node,
  icon: React.Node,
  type: 'success' | 'error' | 'warning',
};

const InfoText = ({ text, icon, type }: Props) => {
  return (
    <MainContainer options={{ type }}>
      <IconContainer>{icon}</IconContainer>
      <BodyContainer>{text}</BodyContainer>
    </MainContainer>
  );
};

export default InfoText;
