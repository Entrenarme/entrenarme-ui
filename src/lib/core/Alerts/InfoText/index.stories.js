// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faAlarmClock,
  faAmbulance,
  faAddressBook,
} from '@fortawesome/pro-light-svg-icons';

import InfoText from './index';

storiesOf('Alerts/Info text', module).add(
  'success type',
  withInfo('success type')(() => (
    <InfoText
      text="success text"
      icon={<FontAwesomeIcon icon={faAlarmClock} />}
      type="success"
    />
  )),
);

storiesOf('Alerts/Info text', module).add(
  'error type',
  withInfo('error type')(() => (
    <InfoText
      text="error text"
      icon={<FontAwesomeIcon icon={faAmbulance} />}
      type="error"
    />
  )),
);

storiesOf('Alerts/Info text', module).add(
  'warning type',
  withInfo('warning type')(() => (
    <InfoText
      text="warning text"
      icon={<FontAwesomeIcon icon={faAddressBook} />}
      type="warning"
    />
  )),
);
