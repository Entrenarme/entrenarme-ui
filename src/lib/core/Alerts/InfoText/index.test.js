import * as React from 'react';
import { render, getByText } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAlarmClock } from '@fortawesome/free-solid-svg-icons';

import InfoText from './index';

const text = 'Test text';

test('renders "InfoText"', () => {
  const { container } = render(
    <InfoText
      text={text}
      icon={<FontAwesomeIcon icon={faAlarmClock} />}
      type="success"
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('renders "InfoText" and check text', () => {
  const { container } = render(
    <InfoText
      text={text}
      icon={<FontAwesomeIcon icon={faAlarmClock} />}
      type="success"
    />,
  );

  getByText(container, text);
});
