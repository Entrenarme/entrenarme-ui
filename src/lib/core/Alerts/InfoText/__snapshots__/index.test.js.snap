// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "InfoText" 1`] = `
<div
  class="sc-bdVaJa bjIpSr"
>
  <div
    class="sc-bwzfXH kYPaus"
  />
  <div
    class="sc-htpNat jnOBXG"
  >
    Test text
  </div>
</div>
`;
